okio/src/commonMain/kotlin/okio/Channel.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2025 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+/**
+ * A pair of streams for interactive communication with another machine.
+ *
+ * Send data to the peer by writing to [sink], and read data from the peer by reading from [source].
+ *
+ * This can be implemented by a plain TCP socket. It can also be layered to add features like
+ * security (as in a TLS channel) or connectivity (as in a proxy channel).
+ *
+ * Closing the [source] does not impact the [sink], and vice versa.
+ */
+interface Channel {
+  val source: Source
+  val sink: Sink
+
+  /**
+   * Fail any in-flight and future operations. After canceling:
```

Comment:
Comparing with https://github.com/square/okhttp/blob/a1863a71f476558389a2f855ca12b34c8eb2a118/mockwebserver/src/main/kotlin/mockwebserver3/internal/duplex/RealStream.kt#L30

That seems have an orderly shutdown, while this seems immediately disruptive.  Mainly because of the "in-flight" statement.

Is that intended?